src/main/scala/tech/cryptonomic/conseil/routes/Tezos.scala

```diff
@@ -70,55 +70,63 @@ object Tezos extends LazyLogging {
   val route: Route = pathPrefix(Segment) { network =>
     get {
       gatherConseilFilter{ filter =>
-        pathPrefix("blocks") {
-          pathEnd {
-            ApiOperations.fetchBlocks(filter) match {
-              case Success(blocks) => complete(JsonUtil.toJson(blocks))
-              case Failure(e) => failWith(e)
-            }
-          } ~ path("head") {
-            ApiOperations.fetchLatestBlock() match {
-              case Success(block) => complete(JsonUtil.toJson(block))
-              case Failure(e) => failWith(e)
-            }
-          } ~ path(Segment) { blockId =>
-            ApiOperations.fetchBlock(blockId) match {
-              case Success(block) => complete(JsonUtil.toJson(block))
-              case Failure(e) => failWith(e)
-            }
-          }
-        } ~ pathPrefix("accounts") {
-          pathEnd {
-            ApiOperations.fetchAccounts(filter) match {
-              case Success(accounts) => complete(JsonUtil.toJson(accounts))
-              case Failure(e) => failWith(e)
-            }
-          } ~ path(Segment) { accountId =>
-            ApiOperations.fetchAccount(accountId) match {
-              case Success(account) => complete(JsonUtil.toJson(account))
-              case Failure(e) => failWith(e)
+        validate(!filter.limit.isDefined || (filter.limit.isDefined && (filter.limit.get <= 10000)), s"Cannot ask for more than 10000 entries") {
```

Comment:
IIRC, we currently allow the user not to provide a `limit` predicate and default to some number when a limit is not provided. Let's not change that behaviour. The point here is to merely prevent the user from asking for too many rows.

Reply:
yes, exactly. !filter.limit.isDefined means that it's okay that the user does not define a limit in their query. if that fails, then it checks if it is defined and it's less than or equal to 10000

Reply:
LOL, I missed that you are testing for a positive, not a negative!